src/interactivity/base/InteractivityFactory.cpp

```diff
@@ -381,4 +390,79 @@ using namespace Microsoft::Console::Interactivity;
 
     return status;
 }
+
+void InteractivityFactory::SetPseudoWindowCallback(std::function<void(bool)> func)
+{
+    _pseudoWindowMessageCallback = func;
+}
+
+[[nodiscard]] LRESULT CALLBACK InteractivityFactory::s_PseudoWindowProc(_In_ HWND hWnd, _In_ UINT Message, _In_ WPARAM wParam, _In_ LPARAM lParam)
+{
+    // Save the pointer here to the specific window instance when one is created
+    if (Message == WM_CREATE)
+    {
+        const CREATESTRUCT* const pCreateStruct = reinterpret_cast<CREATESTRUCT*>(lParam);
+
+        InteractivityFactory* const pFactory = reinterpret_cast<InteractivityFactory*>(pCreateStruct->lpCreateParams);
+        SetWindowLongPtrW(hWnd, GWLP_USERDATA, reinterpret_cast<LONG_PTR>(pFactory));
+    }
+
+    // Dispatch the message to the specific class instance
+    InteractivityFactory* const pFactory = reinterpret_cast<InteractivityFactory*>(GetWindowLongPtrW(hWnd, GWLP_USERDATA));
+    if (pFactory != nullptr)
+    {
+        return pFactory->PseudoWindowProc(hWnd, Message, wParam, lParam);
+    }
+
+    // If we get this far, call the default window proc
+    return DefWindowProcW(hWnd, Message, wParam, lParam);
+}
+
+[[nodiscard]] LRESULT CALLBACK InteractivityFactory::PseudoWindowProc(_In_ HWND hWnd, _In_ UINT Message, _In_ WPARAM wParam, _In_ LPARAM lParam)

```

Comment:
If you all want me to, I'll pull the `PseudoWindowProc` and window creation and whatnot into its own class file. But if you all don't care... then I'll leave it here.

Reply:
I don't care enough at this point. Leaving it here.

Reply:
I don't care enough at this point. Leaving it here.